### ISSUE TITLE
Ensure each logged message has a request id

### ISSUE THREAD

**Issue by PureW** [valhalla/valhalla#2063]
By logging a unique request id with each log-line, it becomes much easier to collect all information of interest when debugging a problematic request that's interleaved with regular traffic in production.

**Comment by danpaz** [comment]
Similar to https://github.com/valhalla/valhalla/issues/1790. Agree this would be very helpful for debugging.

**Comment by enkeyz** [comment]
Just found this: so, because this project already using Boost, can I suggest using Boost/Uuid library?

**Comment by kevinkreiser** [comment]
yeah so at the moment we already do log the request id. it is unique to the running instance (process) but if you are running in a cluster (ie many valhalla processes taking request round-robbin or something behind a load balancers) AND all your logs end up in the same place then i could see why you would want a uniq id across the cluster.

yes we could generate a uuid but i would suggest to do that in prime_server where the request serving path starts. at the moment the ids are just a sequential 64bit id. we could generate a uuid and turn that into 2 64bit numbers to store in the request information but that would require us to write some special hashing functions. the other option is to not do it in prime_server and lose the ability to see when the request came in and when it was returned as a response to the client.

i think the former is much hairier to be honest but i do think its the right thing to do. can we get by with a 64bit random hash rather than a 122bit one? if so then the changes will hopefully be minimal as i dont believe there is anywhere that we rely on sequential ids, though i have to double check. at that point then we just add an option to prime_server to use sequential or random request ids (random isnt free so the option is welcome)

**Comment by PureW** [comment]
As an option it'd also be nice to optionally specify a request-id from whoever is downstream. For example, a CDN sitting in-front often attaches a header with a per request id which would be great to re-use in Valhalla logging. That would allow for a complete picture of a request moving through the pipeline.

**Comment by kevinkreiser** [comment]
in that case i think we can pretty much expect it to be any string of any length... which would require a decent amount of code change but is still do-able. i guess in that case we tell the server what header should be used as the request id

**Comment by yukta8** [comment]
Hello! I came across this issue and found it interesting. Is this still open for contributors to work on? 
If so, I'd be happy to take it up and submit a PR. Please let me know how I can proceed
Thanks!

**Comment by Akhandcode** [comment]
"Hi! I'm a first-time contributor and would like to work on this issue. Could you please assign it to me?"

**Comment by nilsnolde** [comment]
Na, this was mbx times. The current maintainers don't run a commercial api.

Doesn't mean we don't want this. but new contribs can't possibly.
